crates/cheatcodes/src/inspector.rs

```diff
@@ -1674,17 +1665,27 @@ impl<DB: DatabaseExt + Send> Inspector<DB> for Cheatcodes {
 
             let zk_contract = self
                 .dual_compiled_contracts
-                .find_evm_bytecode(&call.init_code.0)
+                .find_by_evm_bytecode(&call.init_code.0)
                 .unwrap_or_else(|| panic!("failed finding contract for {:?}", call.init_code));
 
+            let factory_deps = self.dual_compiled_contracts.fetch_all_factory_deps(zk_contract);
+
+            // get the current persisted factory deps to pass to zk create
+            let persisted_factory_deps = self.persisted_factory_deps.clone();
+            // and extend it for future calls
+            self.persisted_factory_deps
+                .extend(factory_deps.iter().map(|dep| (hash_bytecode(dep), dep.clone())));
```

Comment:
As a follow up, we should try to remove this part, and perhaps extend it only with the `bytecodes` returned from the era vm. So in theory, every known bytecode will reside here. Or perhaps we can also think if it's easier to support it directly in the `Backend` and via some changes to `DatabaseExt` to actually store the factory_deps directly in the storage. But that all can be a later refactor.